openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -518,14 +524,28 @@ class SequenceMetadata(DeveloperErrorViewMixin, APIView):
         SessionAuthenticationAllowInactiveUser,
     )
 
-    def get(self, request, usage_key_string, *args, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
+    def get(self, request, usage_key_or_hash, **kwargs):
         """
         Return response to a GET request.
         """
+        # Load requested sequence from learning_sequences, trying to interpret
+        # the path param first as a UsageKey, and as then as a hash of a UsageKey.
         try:
-            usage_key = UsageKey.from_string(usage_key_string)
-        except InvalidKeyError:
-            raise NotFound(f"Invalid usage key: '{usage_key_string}'.")  # lint-amnesty, pylint: disable=raise-missing-from
+            # Try to parse as a real UsageKey, and then load the sequence.
+            usage_key = UsageKey.from_string(usage_key_or_hash)
+            learning_sequence = get_learning_sequence(usage_key)
+        except (InvalidKeyError, LearningSequenceData.DoesNotExist):
+            learning_sequence = None
+        if not usage_key:
+            # If we are here, then _parsing_ failed.
+            try:
+                # Try treating the path parameter as a *hash* of a UsageKey.
+                learning_sequence = get_learning_sequence_by_hash(usage_key_or_hash)
+                usage_key = learning_sequence.usage_key
+            except LearningSequenceData.DoesNotExist:
+                learning_sequence = None
+        if not learning_sequence:
+            raise NotFound(f"Invalid usage key: '{usage_key_or_hash}'.")
```

Comment:
New behavior 1: If the Sequence Metadata API gets a _hashed_ usage_key as its input, it will leverage learning_sequences to recover the correct sequence and its original usage_key.

This way, the courseware frontend can immediately call the Sequence Metadata API with a hashed usage_key, instead of being blocked on the Course Metadata API's response.